### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by nickelc**

### CODE REVIEW

File: src/book/book.rs [hunk 1]
```diff
@@ -77,10 +77,10 @@ fn create_missing(src_dir: &Path, summary: &Summary) -> Result<()> {
 /// [`iter()`]: #method.iter
 /// [`for_each_mut()`]: #method.for_each_mut
 #[derive(Debug, Clone, Default, PartialEq, Serialize, Deserialize)]
+#[non_exhaustive]
 pub struct Book {
     /// The sections in this book.
     pub sections: Vec<BookItem>,
-    __non_exhaustive: (),
```
Comment (ehuss) [inline src/book/book.rs:83]: This is a breaking change, can you revert it please?

**Review by ehuss — COMMENTED** [review]
Can you maybe put the different lints into different commits so that it is easier to review?

File: src/book/book.rs (hunk 1 above)
Comment (nickelc) [reply src/book/book.rs:83]: Isn't the added `#[non_exhaustive]` attribute the same as the private field?

Comment (Dylan-DPC) [reply src/book/book.rs]: No, it is exposed to preprocessors. There was a similar attempt earlier which was reverted: https://github.com/rust-lang/mdBook/pull/1848

Comment (nickelc) [reply src/book/book.rs]: I removed the change.

**Comment by ehuss** [comment]
Closing as I have applied some clippy fixes in #2083.

**Pull request closed without merging**